src/App.tsx

```diff
@@ -1,14 +1,83 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { FilterStatus } from './types/filter';
+import { debounce } from './utils/debounce';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectTodos, setSelectTodos] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState(FilterStatus.All);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const filteredTodos = useMemo(() => {
+    let filteredTodo = todos;
+
+    if (filter === FilterStatus.Active) {
+      filteredTodo = filteredTodo.filter(todo => !todo.completed);
+    }
+
+    if (filter === FilterStatus.Completed) {
+      filteredTodo = filteredTodo.filter(todo => todo.completed);
+    }
+
+    if (appliedQuery) {
+      filteredTodo = filteredTodo.filter(todo =>
+        todo.title.toLowerCase().includes(appliedQuery.toLowerCase().trim()),
+      );
+    }
+
+    return filteredTodo;
+  }, [todos, filter, appliedQuery]);
+
+  {
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }
+
+  const applyDebounceQuery = useCallback(
+    debounce((value: string) => {
+      setAppliedQuery(value);
+    }),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    applyDebounceQuery(event.target.value);
+    setQuery(event.target.value);
+  };
+
+  useEffect(() => {
+    getTodos().then(todosFromServer => {
+      setTodos(todosFromServer);
+    });
+  }, []);
+
+  const handleShowModal = (userId: number) => {
+    const foundTodo = todos.find(todo => todo.id === userId);
+
+    if (foundTodo) {
+      setSelectTodos(foundTodo);
+    }
+  };
+
+  function changeStatusOfTodos(value: FilterStatus) {
```

Comment:
It is important to follow the same way of creating functions everywhere
```suggestion
  const changeStatusOfTodos = (value: FilterStatus) => {
```